tests/casp/nfs_client.pm

```diff
@@ -0,0 +1,160 @@
+    my $script           = "";
+
+    $script .=
+"cp -a /etc/$etc_file $etc_overlay_path 2> /dev/null || touch $etc_overlay_path/$etc_file\n";
+    $script .= "echo -e \"$what\" >> $etc_overlay_path/$etc_file\n";
+    $script .= "mount -o remount /etc\n";
+
+    return $script;
+}
+
+sub chkdir_strip {
+    my ($nfs_share_dir) = @_;
+    my $script = "";
+
+    $script .=
+"[ \"\$(ls -w1 -A $nfs_share_dir 2> /dev/null | wc -l)\" -gt 0 ] && { echo \"OK, dir $nfs_share_dir not empty\"; true; } || { echo \"FAIL, $nfs_share_dir empty or does not exist\"; false; }\n";
+
+    return $script;
+}
+
+sub run() {
+    my $nfs_uri = get_var('NFS_SHARE');
+
+# format $nfs_uri for mount command similarly to function bootmenu_network_setup() in bootloader_setup.pm
+    $nfs_uri =~ /^nfs:\/\/([\w.-]+)(\/.+)$/;
+    my ( $nfs_server, $nfs_path ) = ( $1, $2 );
+    my $nfs_remotetarget = $nfs_server . ":" . $nfs_path;
+
+    my @nfs_versions = qw(nfs nfs4);
+
+    # Basic remote NFS server checks
+    my $script = "";
+    $script .=
+      "echo -n \$(rpcinfo -s $nfs_server) | egrep 'nfs.*mountd|mountd.*nfs'\n";
+    $script .=
+"echo -n \$(rpcinfo -u $nfs_server nfs) | egrep 'version 3.*version 4|version 4.*version 3'\n";
+    $script .=
+"echo -n \$(rpcinfo -t $nfs_server nfs) | egrep 'version 3.*version 4|version 4.*version 3'\n";
+    $script .= "showmount -e $nfs_server\n";
+
+    print "$script\n";
+    script_output($script);
+
+# TODO as we don't know used NFS_SHARE value we should find some file from the share automatically and copy it to /tmp
+# for eg. by # find /tmp/nfs -type f -size +1M -size -10M -print -quit 2> /dev/null (print path of file >1MB and <10MB and quit)
+# NFSv4 uses different remotetarget with stripped path taken from showmount -e $nfs_server export
+
+    # Autofs NFS test
+    foreach my $a (@nfs_versions) {
+        my $script = "";
+        $script .= "mkdir -p /tmp/$a\n";
+        if ( $a eq "nfs" ) {
+            $script .=
+              casp_etc_strip( "auto.nfs", "share\\t$nfs_remotetarget" );
+        }
+        elsif ( $a eq "nfs4" ) {
+            $script .= casp_etc_strip( "auto.nfs",
+                "share\\t-fstype=nfs4\\t$nfs_server:/" );
+        }
+        $script .= casp_etc_strip( "auto.master",
+            "/tmp/$a\\t/etc/auto.nfs\\t--timeout=10" );
+        $script .= "systemctl start autofs\n";
+        $script .= chkdir_strip("/tmp/$a/share");
+
+        # autofs mount should be unmounted after 10 sec automatically
+        $script .= "sleep 15\n";
+        $script .= "[ \"\$(mount -t $a | wc -l)\" -eq 0 ] && true || false\n";
+        $script .= "systemctl stop autofs\n";
+
+        # Cleanup
+        $script .= "rm /var/lib/overlay/etc/auto.master\n";
+        $script .= "rm /var/lib/overlay/etc/auto.nfs\n";
+        $script .= "mount -o remount /etc\n";
+        $script .= "rm -r /tmp/$a\n";
+
+        print "$script\n";
+        script_output($script);
```

Comment:
I like script_output for its speed and checking of the return codes all the time - it is way faster than anything else.

assert_script_run will write into console for ever... And If I use assert_script_run for each line of the $script then it will produce lot of mess in form of graphical boxes in the openqa web test output (and it will also take forever - I already tried). So if you don't object I would leave it as it is.

I agree with you that my method is not the nicest one but EOF method has also its limits :)